src/main/java/software/amazon/documentdb/jdbc/DocumentDbConnection.java

```diff
@@ -65,14 +66,18 @@ public class DocumentDbConnection extends Connection
     private MongoClient mongoClient = null;
     private MongoDatabase mongoDatabase = null;
     private DocumentDbSshTunnelClient sshTunnelClient;
+    private DocumentDbMetadataService metadataService;
 
     /**
      * DocumentDbConnection constructor, initializes super class.
      */
-    DocumentDbConnection(final DocumentDbConnectionProperties connectionProperties)
+    DocumentDbConnection(
+            final DocumentDbConnectionProperties connectionProperties,
+            final DocumentDbMetadataService metadataService)
             throws SQLException {
         super(connectionProperties);
         this.connectionProperties = connectionProperties;
+        this.metadataService = metadataService;
```

Comment:
I think we might have to check for nulls since the public interface from `DocumentDbDriver` might return a `null` from the `Supplier`.

Reply:
Would it be sufficient to add a not null check in the `DocumentDbDriver` constructor so that the `DocumentDbMetadataService` is never null?